### ISSUE TITLE
Support for MultiDiscrete action spaces

### ISSUE THREAD

**Issue by timmeinhardt** [ikostrikov/pytorch-a2c-ppo-acktr#44]
Currently only _Box_ and _Discrete_ action spaces are supported. For my purposes I would need _MuliDiscrete_ spaces. Is this something that is already in planning? If not I would be happy to implement the missing distribution. The implementation should be straight forward and could be based on the existing `Categorical` distribution. Or am I missing a crucial blocking property of the current implementation that complicates such an endeavour?

**Comment by ikostrikov** [comment]
There are no plans currently to add support for MultiDiscrete actions.

But I will gladly accept a PL!

I'm not sure whether there are any blocking factors but MultiDiscrete actions are clearly useful so I can help to fix any problems.

**Comment by codeislife99** [comment]
@timmeinhardt  
You can use this piece of code
You have to add this in the model.py file
`elif action_space.__class__.__name__ == "MultiBinary":
            num_outputs = action_space.shape[0]
            self.dist = Bernoulli2(self.base.output_size, num_outputs)
`
You have to add this in the distributions.py file
`FixedBernoulli = torch.distributions.Bernoulli
log_prob_bernoulli = FixedBernoulli.log_prob
FixedBernoulli.log_probs = lambda self, actions: log_prob_bernoulli(self, actions).sum(-1, keepdim = True)
entropy = FixedNormal.entropy
FixedNormal.entropy = lambda self: entropy(self).sum(-1)
class Bernoulli2(nn.Module):
    def __init__(self, num_inputs, num_outputs):
        super(Bernoulli2, self).__init__()

        init_ = lambda m: init(m,
              nn.init.orthogonal_,
              lambda x: nn.init.constant_(x, 0),
              gain=0.01)

        self.linear = init_(nn.Linear(num_inputs, num_outputs))

    def forward(self, x):
        x = self.linear(x)
        return FixedBernoulli(logits = x)
`

**Comment by ShaniGam** [comment]
@codeislife99 Thank you for this! 
Do you know how to implement `FixedBernoulli.mode`?

**Comment by codeislife99** [comment]
@ShaniGam. For implementing a Bernoulli mode you have to choose your own threshold probability for each action. During testing, it would be better to just sample instead of using mode. If it's a stochastic environment then mode would not even be close to what you want.
